darts/timeseries.py

```diff
@@ -2019,6 +2142,32 @@ def with_values(self, values: np.ndarray) -> "TimeSeries":
 
         return self.__class__(new_xa)
 
+    def set_static_covariates(
+        self, covariates: Optional[Union[pd.Series, pd.DataFrame]]
+    ):
+        raise_if(
+            not isinstance(covariates, (pd.Series, pd.DataFrame))
+            and covariates is not None,
+            "`covariates` must be either a pandas Series, DataFrame or None",
+            logger,
+        )
+        # check if valid static covariates for multivariate TimeSeries
+        if isinstance(covariates, pd.DataFrame):
+            n_components = len(covariates.columns)
+            raise_if(
+                n_components > 1 and n_components != self.n_components,
+                "When passing a multi-column pandas DataFrame, the number of columns must match the number of "
+                "components of the TimeSeries object (multivariate static covariates must map to each TimeSeries "
+                "component).",
+                logger,
+            )
+        elif isinstance(covariates, pd.Series):
+            covariates = covariates.to_frame()
+        self._xa.attrs["static_covariates"] = (
+            covariates.astype(self.dtype) if covariates is not None else covariates
```

Comment:
I think we could drop this type conversion and later datasets and/or models deal with it downstream - assuming some models such as LGBM can support for instance string types, while others (such as torch models) cannot (we could add a type check with an informative message in the datasets).